examples/vision/zero_dce.py

```diff
@@ -0,0 +1,461 @@
+
+The DCE-Net is a lightweight deep neural network that learns the mapping between an input
+image and its best-fitting curve parameter maps. The input to the DCE-Net is a low-light
+image while the outputs are a set of pixel-wise curve parameter maps for corresponding
+higher-order curves. It is a plain CNN of seven convolutional layers with symmetrical
+concatenation. Each layer consists of 32 convolutional kernels of size 3×3 and stride 1
+followed by the ReLU activation function. The last convolutional layer is followed by the
+Tanh activation function, which produces 24 parameter maps for 8 iterations, where each
+iteration requires three curve parameter maps for the three channels.
+
+![](https://i.imgur.com/HtIg34W.png)
+"""
+
+
+def build_dce_net():
+    input_img = keras.Input(shape=[None, None, 3])
+    conv1 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(input_img)
+    conv2 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv1)
+    conv3 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv2)
+    conv4 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv3)
+    int_con1 = layers.Concatenate(axis=-1)([conv4, conv3])
+    conv5 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con1)
+    int_con2 = layers.Concatenate(axis=-1)([conv5, conv2])
+    conv6 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con2)
+    int_con3 = layers.Concatenate(axis=-1)([conv6, conv1])
+    x_r = layers.Conv2D(24, (3, 3), strides=(1, 1), activation="tanh", padding="same")(
+        int_con3
+    )
+    return keras.Model(inputs=input_img, outputs=x_r)
+
+
+"""
+## Loss Functions
+
+To enable zero-reference learning in DCE-Net, we would use a set of differentiable
+non-reference losses that allow us to evaluate the quality of enhanced images.
+"""
+
+"""
+### Color Constancy Loss
+
+The Color Constancy Loss is used to correct the potential color deviations in the
+enhanced image and also build the relations among the three adjusted channels.
+"""
+
+
+def color_constancy_loss(x, **kwargs):
+    mean_rgb = tf.reduce_mean(x, kwargs.get("axis", (1, 2)), keepdims=True)
+    mr, mg, mb = mean_rgb[:, :, :, 0], mean_rgb[:, :, :, 1], mean_rgb[:, :, :, 2]
+    d_rg = tf.square(mr - mg)
+    d_rb = tf.square(mr - mb)
+    d_gb = tf.square(mb - mg)
+    return tf.sqrt(tf.square(d_rg) + tf.square(d_rb) + tf.square(d_gb))
+
+
+"""
+### Exposure loss
+
+To restrain under-/over-exposed regions, the Exposure Control Loss is used. The exposure
+control loss measures the distance between the average intensity value of a local region
+to the well-exposedness level which is set to `0.6`.
+"""
+
+
+def exposure_loss(x, **kwargs):
+    x = tf.reduce_mean(x, kwargs.get("axis", 3), keepdims=True)
```

Comment:
Likewise here

Reply:
I used keyword args for better readability.